examples/getitem/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+// This is a very fake example of how to check __getitem__ parameter and handle appropriately
+use pyo3::prelude::*;
+use pyo3::exceptions::PyTypeError;
+
+
+#[pyclass]
+struct ExampleContainer {
+
+}
+
+#[pymethods]
+impl ExampleContainer {
+    #[new]
+    fn new() -> Self {
+        ExampleContainer{}
+    }
+
+    fn __getitem__(&self, key: Py<PyAny>, _py: Python) -> PyResult<i32>{
+        // test
+        let key_ref: &PyAny = &key.as_ref(_py);
```

Comment:
I think you can just take `key: &PyAny` as the function argument.

Reply:
Ah, sorry I missed how to do that when building this, switched it to use `&PyAny` instead (which means I no longer have to have the GIL for this particular function (?) which is great).

Reply:
> which means I no longer have to have the GIL for this particular function (?) which is great).

No, the GIL is always held when you call into a `#[pyfunction]`, for whole duration of that call. This just means that PyO3 automatically does that `as_ref` call for you. (And that you do not need the GIL token, but this can always be recovered from a `&PyAny` via [`PyAny::py`](https://docs.rs/pyo3/latest/pyo3/types/struct.PyAny.html#method.py).)